src/Common.Lib/systemstring.h

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// Provides interfaces for working with cross-process strings.
+
+#pragma once
+
+#include "../Common.Headers/tstring.h"
+#ifdef PLATFORM_UNIX
+#include <iconv.h>
+#define _WCHARBASE(wstr)
+
+// On Linux, just call the base constructor for utf strings.
+#define _CHARBASE(str) : osstring(str) {}
+#else
+
+// On Windows, just call the base constructor for wide strings.
+#define _WCHARBASE(wstr) : osstring(wstr) {}
+#define _CHARBASE(str)
+#endif
+
+namespace CommonLib
+{
+#ifdef PLATFORM_UNIX
+    using osstring = std::string;
+#else
+    using osstring = std::wstring;
+#endif
+
+    /// <summary>
+    /// Class for converting between operating system strings and standard strings
+    /// or wide-character (Windows) strings. Useful for converting strings between
+    /// Linux and Windows.
+    /// </summary>
+    class SystemString : public osstring
```

Comment:
Could this be done with a typedef similar to tstring instead of inheritence? (And you could potentially expand the set of static functions on CEncoding or just make all the conversion functions on this class static). 
Imo having a string class with an error state is pretty confusing.